src/lib/services/email-service.ts

```diff
@@ -0,0 +1,173 @@
+import { createTransport, SentMessageInfo, Transporter } from 'nodemailer';
+import Mustache from 'mustache';
+import path from 'path';
+import { readFileSync, existsSync } from 'fs';
+import { Logger, LogProvider } from '../logger';
+import NotFoundError from '../error/notfound-error';
+
+export type AuthOptions = {
+    user: string;
+    password: string;
+};
+
+export enum TemplateFormat {
+    HTML = 'html',
+    PLAIN = 'plain',
+}
+
+export enum TransporterType {
+    SMTP = 'smtp',
+    JSON = 'json',
+}
+
+export type EmailOptions = {
```

Comment:
why is this a type and not an interface?

Reply:
Will revert it back to an interface